### PR TITLE
compiler updated & deployment script improvement

### PR DESCRIPTION
**Pull request by salmanferrum**
This pull request introduces an update in our deployRouter script that has been updated to facilitate the deployment of the OneInchDecoder Library. This library is then linked to the FiberRouter contract.

**Summary:**
- Updated deployRouter script for deploying the OneInchDecoder Library and linking it to FiberRouter contract.

**Changes:**
- Updated deployRouter script for OneInchDecoder Library deployment and linking.

**Rationale:**
The script update ensures the seamless deployment of the OneInchDecoder Library and its linkage to the FiberRouter contract.

**Impact:**
- Enhanced deploy script with the deployment of the OneInchDecoder Library linked to the deployed version of FiberRouter contract.

### CODE REVIEW

**Review by naiemk — APPROVED** [review]

File: contracts/upgradeable-Bridge/FiberRouter.sol [hunk 1]
```diff
@@ -301,74 +341,57 @@ contract FiberRouter is Ownable, TokenReceivable {
...
+    bytes memory oneInchData,
+    address foundryToken,
+    bytes32 withdrawalData,
+    uint256 gasFee
+) external payable {
+    uint256 amountIn = msg.value - gasFee;
+    // Validation checks
+    require(amountIn != 0, "FR: Amount in must be greater than zero");
+    require(gasFee != 0, "FR: Gas fee must be greater than zero");
+    require(msg.value == amountIn + gasFee, "FR: msg.value must equal amountIn plus gasFee");
```
Comment (naiemk) [inline contracts/upgradeable-Bridge/FiberRouter.sol:360]: Redundant

**Review by naiemk — CHANGES_REQUESTED** [review]
Why not using FerrumDeployer pattern to deploy contracts?

File: contracts/upgradeable-Bridge/FiberRouter.sol (hunk 1 above)
Comment (salmanferrum) [reply contracts/upgradeable-Bridge/FiberRouter.sol:360]: the deployer pattern is already implemented and a PR is also created by @SibghatUllah1997 https://github.com/ferrumnet/MultiSwap/pull/39

**Comment by taha-abbasi** [comment]
@salmanferrum please resolve the merge conflicts arising from merge of #39

**Comment by salmanferrum** [comment]
> @salmanferrum please resolve the merge conflicts arising from merge of #39

@taha-abbasi conflicts are resolved.

**Review by taha-abbasi — APPROVED** [review]

**Review by taha-abbasi — APPROVED** [review]

**Pull request merged**